### DISCUSSION TITLE
Web Interface

### DISCUSSION THREAD

**Discussion by peteallenm** [corrados/edrumulus#153] [General]
Hi,

I've done a prototype of a web interface to allow settings and data visualisation. It's heavily based on stuff I've used before, and at the moment is very much in prototype form. I'm aware it's quite a few changes from an unknown person, but would be interested to know if it would be something you think is worthwhile developing further.

I've put a quick video to show what it looks like so far here:
https://youtube.com/shorts/xEUSovF_s58

And the code is on a branch here, but it's very much a prototype at the moment.
I've moved the main processing from loop() to tasks (one for EDrumulus and one for midi).
The web stuff then happens in the idle loops so shouldn't affect drum performance
https://github.com/peteallenm/edrumulus/tree/esp32_s3_12input
Any thoughts?

**Comment by peteallenm** [comment]
I forgot to say- i haven't get got a pad attached to this, so the visualization is just showing noise...

**Reply by kozukioden374** [reply]
Can you help me in setting my esp32 S3. I couldn't connect it to your webui.

I see the edrumulus on the wifi option in my computer but i cannot access it, it always shows the password is wrong

**Reply by peteallenm** [reply]
It's a very long time since I made that branch... I would make sure you've uploaded the webpage (pio run -t uploadfs -e esp32s3usbotg). if that doesn't work I'm not sure. You could always download opencode and get an LLM to fix it for you - It's pretty good for things like this (I'd go deepseek)

**Comment by corrados** [comment]
Looks good. I'm curious how you like the performance of the position detection if you use real hardware.

**Comment by peteallenm** [comment]
I've dug out my snare. I'll be honest-it's diy and it's not the best. I'll need to tune it a fair bit (and probably improve the pad). Positional sensing with the central piezo does work though, but it's not massively consistent. 

https://youtu.be/v19uzAI9OaY

I'll try to wire up 3 piezos and compare.

**Reply by corrados** [reply]
I did also tried out to use a DIY piezo cone on my PD-120 (test with a second piezo) but I got pretty bad results for the positional sensing using that piezo. I think the design of the cone for the piezo may influence the positional sensing performance more than I thought.
